src/account-mgr.cpp

```diff
@@ -207,3 +204,41 @@ Account AccountManager::getAccountBySignature(const QString& account_sig) const
 
     return Account();
 }
+
+void AccountManager::updateServerInfo()
+{
+    for (size_t i = 0; i < accounts_.size(); i++)
+        updateServerInfo(i);
+}
+
+void AccountManager::updateServerInfo(unsigned index)
+{
+    ServerInfoRequest *request;
+    // request is taken owner by Account object
+    request = accounts_[index].createServerInfoRequest();
+    connect(request, SIGNAL(success(const Account&, const ServerInfo &)),
+            this, SLOT(serverInfoSuccess(const Account&, const ServerInfo &)));
+    connect(request, SIGNAL(failed(const ApiError&)),
+            this, SLOT(serverInfoFailed(const ApiError&)));
+    request->send();
+}
+
+
+void AccountManager::serverInfoSuccess(const Account &account, const ServerInfo &info)
+{
+    for (size_t i = 0; i < accounts_.size(); i++) {
+        if (accounts_[i] == account) {
+            if (i == 0)
+                emit beforeAccountChanged();
+            accounts_[i].serverInfo = info;
+            if (i == 0)
+                emit accountsChanged();
```

Comment:
I think we have discussed this topic before and made a decision there. I still don't know why you change your mind and what is so important that make you think using signal `accountInfoUpdated` instead is more correct.

Yes, but I am con to this change which involves new complexities and potential bugs. Anyway, if you insist to do it, I suggest you to do it in a separate PR as kind of improvement. According to your words, it shouldn't cost too much code there, so it is a pretty easy job. Above all be careful that don't introduce new bugs there.